CPP_STYLE_GUIDE.md

```diff
@@ -257,6 +258,24 @@ class ExampleClass {
 };
 ```
 
+### When to use AliasedBuffers
+
+When working with typed arrays that involves direct data modification
+from C++, and when such modifications can have side effects outside
+of the callsite, use an AliasedBuffer when possible, because that
+makes it easier for some JS debuggers to observe these modifications.
+
+```c++
+// Create an AliasedBuffer
+AliasedBuffer<uint32_t, v8::Uint32Array> data;
+...
+
+// Modify the data through natural operator semantics,
+// but those operators are overloaded in AliasedBuffer
+// and provide placeholders to hook mutation observers.
```

Comment:
I think that this provides placeholders is only relevant to VM engine implementers (i.e. not Node.js coders), so I don’t think it needs to be in the style guide.